arrayfire/library/vector_algorithms.py

```diff
@@ -207,9 +207,9 @@ def sum(array: Array, /, *, axis: int | None = None, nan_value: float | None = N
         return wrapper.sum_nan_all(array.arr, nan_value)
 
     if nan_value is not None:
```

Comment:
let's invert this conditional to match the one in product(). It's easier to reason about `is None` `than is not None`, probably why it was mistaken in the first place